src/Microsoft.OData.Core/UriParser/SemanticAst/SelectExpandClauseExtensions.cs

```diff
@@ -106,12 +106,10 @@ internal static void Traverse<T>(this SelectExpandClause selectExpandClause, Fun
             {
                 string currentExpandClause = String.Join("/", expandSelectItem.PathToNavigationProperty.WalkWith(PathSegmentToStringTranslator.Instance).ToArray());
                 T subResult = default(T);
-                if (expandSelectItem.SelectAndExpand.SelectedItems.Any())

```

Comment:
>expandSelectItem.SelectAndExpand.SelectedItems.Any()) [](start = 20, length = 53)

why are you removing this check?  won't traverse always return an empty subresult if there are no items in SelectedItems?

Reply:
restored. Same outcome.

---
In reply to: [224144946](https://github.com/OData/odata.net/pull/1286#discussion_r224144946) [](ancestors = 224144946)